### DISCUSSION TITLE
How to restrict a program of filesystem access?

### DISCUSSION THREAD

**Discussion by CodeCubeNeo** [sandboxie-plus/Sandboxie#2573] [Q&A Feedback]
How do I restrict a program like DiskInternal's LinuxReader, to only one drive (and maybe read only)? And how to only allow it to write in one specific folder on the host?

**Comment by offhub** [comment]
Are you using the Plus UI?

**Comment by CodeCubeNeo** [comment]
yes, sandboxie-plus version if that is what u mean?

**Comment by offhub** [comment]
`Right-click on the sandbox` > `Sandbox Options` > `Resource Access` > `Files` > `Add File/Folder`

![lnxrdr](https://user-images.githubusercontent.com/6871698/209980590-0ed4ba0e-3193-41f7-92e1-308a92726030.png)

- By default Sandboxie doesn't allow writing to Host system.

1. Select `Open` if the program is installed _outside_ the sandbox.
2. Select `Open for All` if the program is installed _inside_ the sandbox.

- You can get the disk numbers from `Windows` > `Run` > `DiskMgmt`
- Uncheck the boxes if you don't want to block disk then `Apply`.

Example config: (Sandbox Options > Edit ini Section)
```
OpenFilePath=%UserProfile%\Downloads
ClosedFilePath=\Device\Harddisk0\
ClosedFilePath=\Device\Harddisk1\
```

**Reply by ghost** [reply]
I was still able to browse folders on this disk. Closing path to partition worked though

**Comment by scscgit** [comment]
This file access UI is so convoluted and user-unfriendly :( Do people generally never protect their files?
